### DISCUSSION TITLE
Why is there no code in the granite-code-models repo?

### DISCUSSION THREAD

**Discussion by Teo-Asinari** [ibm-granite/discussions#4] [Q&A] [answered]
Sorry for the noob question. The [this article](https://www.zdnet.com/article/ibm-open-sources-its-granite-ai-models-and-they-mean-business/) (linked from the HN post) claims it is fully open source. I was wondering if that merely means we can download pre-trained models and weights, or we can actually see the code used to train the models in the first place. Many thanks

**Comment by Teo-Asinari** [comment]
> Sorry for the noob question. The HN post and [this article](https://www.zdnet.com/article/ibm-open-sources-its-granite-ai-models-and-they-mean-business/) claim it is fully open source. I was wondering if that merely means we can download pre-trained models and weights, or we can actually see the code used to train the models in the first place. Many thanks

Just to be clear, it seems there's only the main branch and FWICT there really is no code there, just some docs and a pdf

**Reply by mayank31398** [reply, marked as answer]
Hey @Teo-Asinari 
we are planning on releasing all our finetuning and pretraining code.
Currently, I am in the process of cleaning up the repos.
And they will be released shortly.

**Reply by Teo-Asinari** [reply]
Thanks @mayank31398, I'll keep my eyes peeled!

**Reply by mayank31398** [reply]
https://github.com/ibm-granite/dolomite-engine
this is our codebase for finetuning.
currently, I am working on documenting it.

**Reply by noobmldude** [reply]
Thanks for sharing the finetuning code. Is there a plan to release the pretraining code as well?

**Reply by mayank31398** [reply]
the model was pretrained using https://github.com/open-lm-engine/lm-engine
